executor/linux/secret.go

```diff
@@ -5,212 +5,398 @@
 package linux
 
 import (
+	"bufio"
+	"bytes"
 	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"strings"
+	"time"
 
+	"github.com/drone/envsubst"
 	"github.com/go-vela/types/constants"
 	"github.com/go-vela/types/library"
 	"github.com/go-vela/types/pipeline"
 
 	"github.com/sirupsen/logrus"
 )
 
-// PullSecret defines a function that pulls the secrets for a given pipeline.
-func (c *client) PullSecret(ctx context.Context) error {
-	var err error
+// secretSvc handles communication with secret processes during a build.
+type secretSvc svc
 
-	p := c.pipeline
+var (
+	// ErrUnrecognizedSecretType defines the error type when the
+	// SecretType provided to the client is unsupported.
+	ErrUnrecognizedSecretType = errors.New("invalid secret type")
```

Comment:
Thoughts on this?

```suggestion
	// ErrUnrecognizedSecretType defines the error type when the
	// SecretType provided to the client is unsupported.
	ErrUnrecognizedSecretType = errors.New("unrecognized secret type")
```

I ask because that is what the variable name and comment state so not sure if we want to be more consistent?

Reply:
I was back and forth between invalid and unrecognized. I'm cool with that change, I would like to get this into types somehow at some point.